### DISCUSSION TITLE
My minecraft suddenly stopped starting at 10 AM on May 5

### DISCUSSION THREAD

**Discussion by yamanobe96** [itzg/docker-minecraft-bedrock-server#320] [Q&A]
I have been playing minecraft with docker since january 2023 using a synology NAS!
However, since yesterday I suddenly cannot connect to it.
Why?
This is the output log
SYONOLOGY has output in HTML.
(HTML cannot be uploaded, so it is converted to a TXT file.

[5-5 itzg-minecraft-bedrock-server1.txt](https://github.com/itzg/docker-minecraft-bedrock-server/files/11411905/5-5.itzg-minecraft-bedrock-server1.txt)
[5-6 itzg-minecraft-bedrock-server1.txt](https://github.com/itzg/docker-minecraft-bedrock-server/files/11411906/5-6.itzg-minecraft-bedrock-server1.txt)

This is the exported file
[Downloads.zip](https://github.com/itzg/docker-minecraft-bedrock-server/files/11411909/Downloads.zip)

**Comment by yamanobe96** [comment]
I apologize if my PC knowledge is limited and I don't have enough information.
I'd be glad to know what you need to do to solve the problem.

**Comment by itzg** [comment]
I'm afraid it's going to be something unique to your system. I'm not seeing anything obvious, but those logs are massive so I only scrolled to the end.

Test things out by creating a fresh, temp container using:

```
docker run -it --rm -e EULA=true -p 19132:19132/udp itzg/minecraft-bedrock-server
```

**Comment by yamanobe96** [comment]
Sorry, I tried putting that command into SSH, but it doesn't work!

nasname:/$ docker run -it --rm -e EULA=true -p 19132:19132/udp itzg/minecraft-bedrock-server
docker: Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post "http://%2Fvar%2Frun%2Fdocker.sock/v1.24/containers/create": dial unix /var/run/docker.sock: connect: permission denied.
See 'docker run --help'.


I've removed and reinstalled docker's itzg-minecraft-bedrock-server, but the result is the same
[itzg-minecraft-bedrock-server1.zip](https://github.com/itzg/docker-minecraft-bedrock-server/files/11413657/itzg-minecraft-bedrock-server1.zip)

**Comment by itzg** [comment]
In that case, you need to "sudo" in front of that. 

Set `DEBUG` environment variable to "true" and post those logs.

**Comment by yamanobe96** [comment]
nasname:/$ sudo docker run -it --rm -e EULA=true -p 19132:19132/udp itzg/minecraft-bedrock-server
DEBU[0000] Using /data to match uid and gid
DEBU[0000] Resolved UID=0 from match path
DEBU[0000] Resolved GID=0 from match path
Looking up latest version...

make trouble for someone
Do you understand now?

**Comment by itzg** [comment]
> make trouble for someone
> Do you understand now?

No, I don't understand. Also, I asked for DEBUG to be enabled. Please run

```
sudo docker run -it --rm -e EULA=true -e DEBUG=true -p 19132:19132/udp itzg/minecraft-bedrock-server
```

**Comment by yamanobe96** [comment]
sorry,I've re-run it.

nasname:/$ sudo docker run -it --rm -e EULA=true -e DEBUG=true -p 19132:19132/udp itzg/minecraft-bedrock-server
Password:
DEBU[0000] Using /data to match uid and gid
DEBU[0000] Resolved UID=0 from match path
DEBU[0000] Resolved GID=0 from match path
+ curlArgs=(-v)
++ id -a
++ ls -ld /data
+ echo 'DEBUG: running as uid=0(root) gid=0(root) groups=0(root) with drwxr-xr-x 1 root root 0 Apr 10 18:31 /data'
DEBUG: running as uid=0(root) gid=0(root) groups=0(root) with drwxr-xr-x 1 root root 0 Apr 10 18:31 /data
++ pwd
+ echo '       current directory is /data'
       current directory is /data
+ downloadPage=https://www.minecraft.net/en-us/download/server/bedrock
+ [[ TRUE != TRUE ]]
+ case ${VERSION^^} in
+ echo 'Looking up latest version...'
+ lookupVersion serverBedrockLinux
Looking up latest version...
+ platform=serverBedrockLinux
+ for i in {1..3}
++ restify --user-agent=itzg/minecraft-bedrock-server --headers 'accept-language:*' --attribute=data-platform=serverBedrockLinux https://www.minecraft.net/en-us/download/server/bedrock
++ jq -r '.[0].href'
++ echo ''
+ DOWNLOAD_URL=
+ [[ -n '' ]]
+ sleep 1
+ for i in {1..3}
++ restify --user-agent=itzg/minecraft-bedrock-server --headers 'accept-language:*' --attribute=data-platform=serverBedrockLinux https://www.minecraft.net/en-us/download/server/bedrock
++ jq -r '.[0].href'
++ echo ''
+ DOWNLOAD_URL=
+ [[ -n '' ]]
+ sleep 1
+ for i in {1..3}
++ restify --user-agent=itzg/minecraft-bedrock-server --headers 'accept-language:*' --attribute=data-platform=serverBedrockLinux https://www.minecraft.net/en-us/download/server/bedrock
++ jq -r '.[0].href'
++ echo ''
+ DOWNLOAD_URL=
+ [[ -n '' ]]
+ sleep 1
+ [[ -z '' ]]
++ curl -s https://mc-bds-helper.vercel.app/api/latest
+ DOWNLOAD_URL=
nasname:/$

**Comment by itzg** [comment]
Something is wrong with your docker/system networking. See if this reveals any clues:

```
sudo docker run -it --rm --entrypoint curl itzg/minecraft-bedrock-server -vfsSL -H 'accept-language:*' -A itzg/minecraft-bedrock-server -o /dev/null https://www.minecraft.net/en-us/download/server/bedrock
```

**Comment by yamanobe96** [comment]
Thank you for your response.
I immediately executed it.

nasname:/$ sudo docker run -it --rm --entrypoint curl itzg/minecraft-bedrock-server -vfsSL -H 'accept-language:*' -A itzg/minecraft-bedrock-server -o /dev/null https://www.minecraft.net/en-us/download/server/bedrock
Password:
* Could not resolve host: www.minecraft.net
* Closing connection 0
curl: (6) Could not resolve host: www.minecraft.net
nasname:/$

**Comment by yamanobe96** [comment]
Sorry, resolved.
I just rebooted the synology NAS and the minecraft server worked fine.
Thanks a lot for your help!
I am very happy that it works!

**Comment by itzg** [comment]
I’m going to convert to discussion so it can remain open for others to find similar problem and solution.

**Comment by maddox** [comment]
I'm having issues booting latest with an issue of not being able to resolve the domain where it downloads the latest bedrock server. This was mentioned in #340 

I just ran the command you mentioned on this thread, here are my results:

```
jmaddox@server:~/docker$ docker run -it --rm --entrypoint curl itzg/minecraft-bedrock-server -vfsSL -H 'accept-language:*' -A itzg/minecraft-bedrock-server -o /dev/null https://www.minecraft.net/en-us/download/server/bedrock
* getaddrinfo() thread failed to start
* Could not resolve host: www.minecraft.net
* Closing connection 0
curl: (6) getaddrinfo() thread failed to start
```

This issue is so weird.

**Comment by magnuznilzzon** [comment]
I'm seeing the same issue,

Output from a docker-compose start with ENABLED is here: https://pastebin.com/1Xe5xXdG

bds    | + curl -v -o /tmp/bedrock-server-1.20.13.01.zip -fsSL https://minecraft.azureedge.net/bin-linux/bedrock-server-1.20.13.01.zip
bds    | * getaddrinfo() thread failed to start
bds    | * Could not resolve host: minecraft.azureedge.net

**Reply by magnuznilzzon** [reply]
Ah, some googling on "getaddrinfo() thread failed to start" showedb ([link](https://stackoverflow.com/questions/74959500/getaddrinfo-thread-failed-to-start-while-building-docker-image))  that this probably was due to a mismatch between the docker version and the host OS version. Updating Docker solved my issue :)

**Reply by maddox** [reply]
Wow. I'll give that a try and see what happens.

**Reply by maddox** [reply]
Hmm, I updated and got `version 19.03.13`, and latest is `version 24.x`. My host is running `Ubuntu 20.04.6 LTS`, so I wonder if it's just not getting the latest packages, though I thought the package source was docker itself, not Ubuntu.

What version of docker are you running that resolved this?

**Reply by itzg** [reply]
Take a look at https://docs.docker.com/engine/install/ubuntu/ to make sure you are using the official packages

**Reply by magnuznilzzon** [reply]
Yes, like ITZG says, (kind of) you need to add in the new repository, since that is how Docker works, one repo per major release I think. So I just followed the instructions for a fresh install, from the heading "Install using the apt repository" onwards. 

In the end I got a repository in apt that was named something-jammy instead of something-groovy, and then apt update found a new docker version for me.

**Reply by maddox** [reply]
OK, yes, it appears my original installation method was outdated and was not returning the latest version. I uninstalled and re-installed with the modern method by installing `docker-ce`. This got me on the latest version, and this container now works correctly.
